Providers/nxOMSAgentNPMConfig/WMIAdapter.c

```diff
@@ -0,0 +1,196 @@
+/* @migen@ */
+#include <windows.h>
+#include <stdlib.h>
+#include <mi.h>
+
+#define STR_HELPERDLLNAME L"wmitomi.dll"
+#define STR_REGISTERDLL "Adapter_RegisterDLL"
+#define STR_UNREGISTERDLL "Adapter_UnRegisterDLL"
+#define STR_DLLCANUNLOADNOW "Adapter_DllCanUnloadNow"
+#define STR_DLLGETCLASSOBJECT "Adapter_DllGetClassObject"
+
+typedef MI_Module* (MI_MAIN_CALL * fnMI_Main)(_In_ MI_Server* server);
+typedef HRESULT (__stdcall * fnRegisterDLL)(HINSTANCE module, GUID classId);
+typedef HRESULT (__stdcall * fnUnRegisterDLL)(GUID classId);
+typedef HRESULT (__stdcall * fnDllCanUnloadNow)(void);
+typedef HRESULT (__stdcall * fnDllGetClassObject)(CLSID supportedClassIds,
+    fnMI_Main main,
+    REFCLSID rclsid,
+    REFIID riid,
+    LPVOID * ppv);
+
+// Helper DLL module handle
+volatile HMODULE g_hHelper = NULL;
+
+// Main provider entry point for MI provider. The framework
+// needs to call into this to get all run time type information.
+MI_EXTERN_C MI_Module* MI_MAIN_CALL MI_Main(_In_ MI_Server* server);
+
+// DLL module handle, needed for COM registration
+HINSTANCE g_hModule = NULL;
+
+// Unique provider ID
+// {43EF424B-F149-4B34-9315-4A3BBE8E6696}
+CLSID g_providerClassID = { 0x43ef424b, 0xf149, 0x4b34, { 0x93, 0x15, 0x4a, 0x3b, 0xbe, 0x8e, 0x66, 0x96 } };
```

Comment:
Can you create a new GUID here? There might be a potential issue using the same ClassID for different DSC modules. I've updated the one note for this session.